terraform/azure/modules/hana_node/main.tf

```diff
@@ -364,62 +364,60 @@ locals {
   disks_writeaccelerator = [for writeaccelerator in split(",", var.hana_data_disks_configuration["writeaccelerator"]) : tobool(trimspace(writeaccelerator))]
 }
 
-resource "azurerm_virtual_machine" "hana" {
-  count                            = var.hana_count
-  name                             = "${var.name}${format("%02d", count.index + 1)}"
-  location                         = var.az_region
-  resource_group_name              = var.resource_group_name
-  network_interface_ids            = [element(azurerm_network_interface.hana.*.id, count.index)]
-  availability_set_id              = var.common_variables["hana"]["ha_enabled"] ? azurerm_availability_set.hana-availability-set[0].id : null
-  vm_size                          = var.vm_size
-  delete_os_disk_on_termination    = true
-  delete_data_disks_on_termination = true
-
-  storage_os_disk {
-    name              = "disk-${var.name}${format("%02d", count.index + 1)}-Os"
-    caching           = "ReadWrite"
-    create_option     = "FromImage"
-    managed_disk_type = "Premium_LRS"
-  }
+resource "azurerm_managed_disk" "hana_data_disk" {
+  count                = var.hana_count * local.disks_number
+  name                 = "disk-${var.name}${format("%02d", floor(count.index / local.disks_number) + 1)}-Data${format("%02d", count.index % local.disks_number + 1)}"
```

Comment:
Was
```
name = "${var.name}${format("%02d", count.index + 1)}"
```

why a more complicated disk name is needed?

Reply:
because the name needs to be unique for each disk, and since before we were using a loop for each disk for each VM, now that the loop is gone we have to incorporate it in the variables (count is used to let terraform know how many resources to spawn: https://buildvirtual.net/terraform-count-examples/). Also see answer in the next question